interop-cats/jvm/src/main/scala/scalaz/zio/interop/stm/STM.scala

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio.interop.stm
+
+import cats.~>
+import scalaz.zio.ZIO
+import scalaz.zio.stm.{ STM => ZSTM }
+
+import scala.util.Try
+
+/**
+ * See [[scalaz.zio.stm.STM]]
+ */
+final class STM[F[+ _], +A] private[stm] (private[stm] val underlying: ZSTM[Throwable, A])(
```

Comment:
I'd have this class extend `AnyVal`, which you will be able to do when removing the implicit.

Reply:
I would like to make it AnyVal too, but scala compiler says `value class may not wrap another user-defined value class`, because ZSTM is also an AnyVal. I don't see how to overcome this restriction for now.